internal/generator/vector/source/kubernetes_logs.go

```diff
@@ -36,42 +42,94 @@ rotate_wait_ms = 5000
 {{end}}`
 }
 
-func NewKubernetesLogs(id, namespace string) KubernetesLogs {
+func NewKubernetesLogsForOpenShiftLogging(id string) KubernetesLogs {
+
+	excludes := []string{}
+	for _, comp := range []string{constants.LogfilesmetricexporterName, constants.ElasticsearchName, constants.KibanaName} {
+		comp = fmt.Sprintf(`"/var/log/pods/%s_%s-*/*/*.log"`, constants.OpenshiftNS, comp)
+		excludes = append(excludes, comp)
+	}
+	for _, comp := range []string{"loki*", "gateway", "opa"} {
+		comp = fmt.Sprintf(`"/var/log/pods/%s_*/%s/*.log"`, constants.OpenshiftNS, comp)
+		excludes = append(excludes, comp)
+	}
+	for _, comp := range []string{"gz", "tmp"} {
+		comp = fmt.Sprintf(`"/var/log/pods/*/*/*.%s"`, comp)
+		excludes = append(excludes, comp)
+	}
+	return NewKubernetesLogs(id, "", joinContainerPathsForVector(excludes))
+}
+
+// NewKubernetesLogs element which always excludes temp and gzip files
+func NewKubernetesLogs(id, includes, excludes string) KubernetesLogs {
 	return KubernetesLogs{
 		ComponentID:  id,
 		Desc:         "Logs from containers (including openshift containers)",
-		ExcludePaths: ExcludeContainerPaths(namespace),
+		IncludePaths: includes,
+		ExcludePaths: excludes,
 	}
 }
 
-func LogFilesMetricExporterLogsPath(namespace string) string {
-	return fmt.Sprintf("/var/log/pods/%s_%%s-*/*/*.log", namespace)
-}
+const (
+	crioNamespacePathFmt = `"/var/log/pods/%s/*/*.log"`
+	crioContainerPathFmt = `"/var/log/pods/*/%s/*.log"`
+	crioPathExtFmt       = `"/var/log/pods/*/*/*.%s"`
+)
 
-func ElasticSearchLogStoreLogsPath(namespace string) string {
-	return fmt.Sprintf("/var/log/pods/%s_%%s-*/*/*.log", namespace)
+// ContainerPathGlobFrom formats a list of kubernetes container file paths to include/exclude for
+// collection given a list of namespaces and containers and return a string that
+// is in a form directly usable by a vector kubernetes_log config. The result is
+// a set of file paths assumed to be at the well known location and structure of
+// CRIO pod logs.
+// The format rules:
+//
+//	namespaces:
+//	  namespace:     /var/log/pods/namespace_*/*/*.log
+//	  **namespace:   /var/log/pods/*namespace_*/*/*.log
+//	  **name*pace**: /var/log/pods/*name*pace*/*/*.log
+//	  namespace**:   /var/log/pods/namespace*/*/*.log
+//	containers:
+//	  container:    /var/log/pods/*/container/*.log
+//	  *cont**iner*:    /var/log/pods/*/*cont*iner*/*.log
+//	  cont**iner*:    /var/log/pods/*/cont*iner*/*.log
+func ContainerPathGlobFrom(namespaces, containers []string, extensions ...string) string {
```

Comment:
No implementation I can see.  Is this for testing only, seems it could/should be incorporated.

Reply:
I'm currently writing something now that utilizes this function to enable dedicated inputs to container source config